data/sensors_recent.php

```diff
@@ -292,18 +400,36 @@ function output_cell($rowspan, $data) {
 			} else {
 				
 				$distance = false;
+				#printf("432 row[latitude]=%s gwdata[latitude]=%s<br>", $row['latitude'], $gwdata['latitude']);
 				if ($row['latitude'] && array_key_exists('latitude', $gwdata) && $gwdata['latitude']) {
 					$distance = haversineGreatCircleDistance($row['latitude'], $row['longitude'], $gwdata['latitude'], $gwdata['longitude']);
 				}
+
 				$gw_id = $gwdata['gtw_id'];
-				echo("  <td>" . gateway_button($gw_id, $gwdata) . "</td>\n");
+				if ($gw_id == "unknown") {
+					echo("  <td><I>" . $gw_id . "</i></td>\n");
+				} else if (($gwdata['longitude'] == 0) && ($gwdata['latitude'] == 0)) {
+					echo("  <td>" . $gw_id . "</td>\n");
+				} else {
+					echo("  <td>" . gateway_button($gw_id, $gwdata) . "</td>\n");
+				}
+
 				if ($distance)
-					echo("  <td>" . round($distance / 1000, 3) . "km</td>\n");
+					echo("  <td>" . round($distance / 1000, 3) . " km</td>\n");
 				else
 					echo("  <td>-</td>\n");
+
 				echo("  <td>" . htmlspecialchars($gwdata["rssi"]) . "</td>\n");
-				echo("  <td>" . htmlspecialchars($gwdata["snr"]) . "</td>\n");
-				echo("  <td>" . htmlspecialchars($metadata["frequency"]) . "Mhz, " . htmlspecialchars($metadata["data_rate"]) . ", " .htmlspecialchars($metadata["coding_rate"]) . "CR</td>\n");
+
+				if ($gwdata["snr"] == $null_snr)
+					echo("  <td>x</td>\n");
+				else 
+					echo("  <td>" . htmlspecialchars($gwdata["snr"]) . "</td>\n");
+
+				if ($row['source'] == 'ttn.v2')
+					echo("  <td>" . htmlspecialchars($metadata["frequency"]) . "MHz, " . htmlspecialchars($metadata["data_rate"]) . ", " .htmlspecialchars($metadata["coding_rate"]) . "CR</td>\n");
+				if ($row['source'] == 'ttn.v3')
+					echo("  <td>" . htmlspecialchars($metadata["frequency"]) . "MHz, SF" . htmlspecialchars($metadata["spreading_factor"]) . "BW" . htmlspecialchars($metadata["data_rate"]) . ", " .htmlspecialchars($metadata["coding_rate"]) . "CR</td>\n");
```

Comment:
Ik zou deze conversie (`SF`, `spreading_factor`, `BW` en `data_rate` aan elkaar plakken) bovenaan doen, waar je toch al v3 apart behandeld en het resultaat in `$metadata["data_rate"]` stoppen. Je kunt dan hier gewoon onafhankelijk van de source versie gewoon `$metadata["data_rate"]` direct gebruiken (dus zoals het er origineel stond).